corehq/apps/sso/backends.py

```diff
@@ -0,0 +1,56 @@
+from django.contrib.auth.backends import ModelBackend
+from django.contrib.auth.models import User
+
+from corehq.apps.sso.models import IdentityProvider, AuthenticatedEmailDomain
+
+
+class SsoBackend(ModelBackend):
+    """
+    Authenticates against an IdentityProvider and SAML2 session data.
+    """
+
+    def authenticate(self, request, username=None, idp_slug=None,
+                     is_handshake_successful=False, **kwargs):
```

Comment:
These argument defaults are not necessary
```suggestion
    def authenticate(self, request, username, idp_slug, is_handshake_successful):
```

Django's [`auth.authenticate` function](https://github.com/django/django/blob/stable/2.2.x/django/contrib/auth/__init__.py#L67-L71) will not try to call this method if the supplied `**credentials` do not match the arguments required here. In fact, it is more efficient to make them required (and not accept `**kwargs`) because it will cause the method to _not_ be called on other non-SSO authentication attempts. As an example, note how [`RemoteUserBackend`](https://github.com/django/django/blob/stable/2.2.x/django/contrib/auth/backends.py#L128) is implemented with no defaults, and it also extends `ModelBackend`.